_posts/2020-03-22-worldwide-rse-mentoring-network.md

```diff
@@ -0,0 +1,240 @@
+[tidy data](https://vita.had.co.nz/papers/tidy-data.pdf)
+format whenever I can, even for something that feels less like an
+analysis and more like "data munging", e.g. splitting a "big" dataset into
+training and test sets for machine learning.
+Before I woke up to the beauty of tidy data, I would spin my wheels doing a lot
+of things like writing functions to pack and unpack Numpy arrays into Python
+dictionaries. Which led me to do things like sharing my data by
+dumping a bunch of Numpy arrays into .json files. That wasn't *wrong*
+necessarily, but it's not *great*. Especially when I could have replaced all my
+bespoke handmade data dumping functions with a simple `df.to_csv()`.
+This brilliant insight only came to me after finding out the hard way that
+my other approaches were not so great, especially when I had to drill down
+on unexpected results and sniff out the source of bugs. It also mainly came to
+me from cross pollinating with R people who almost by default learn to think that
+way (assuming they converted to R after the advent of the Tidyverse).
+
+So maybe you can see why it seems to me there has to a better way to help
+scientist-coders learn. Even the dumb ones like me.
+I guess I'm asking, are there things we can do as a (PyOpenSci) community
+to further help other scientist-coders avoid my pain. And not just general
+stuff like what are best practices for cleaning and transforming data. I'm
+almost more worried about things that are domain specific. How can we help a
+relative newbie answer questions like:
+what libraries do people in our domain usually use?
+And also questions that feel more Python specific, like:
+When should I use sub-packages?
+Why would I ever want an `__all__` statement in my module?
+
+Again, if you are reading this blog, there's a good chance you know exactly
+the right combination of search terms to start to answer those questions.
+Given half an hour, you could find a handful of
+good examples of, say, GitHub repositories associated with papers in
+your domain of science. I know I could do the same. (Replace GitHub repo
+with Zenodo entry, or OSI project, or whatever your preferred flavor of open is.)
+*But* there's probably a ton of expert knowledge you've built up, and
+maybe kind of even forgotten that you have, that guides you as you select those
+handful of good examples. **How do we get most efficiently get scientists
+that are *relatively new* to writing research code to that level of expertise?**
+
+## Straw Man Solutions
+
+First, let me say I am fully aware there are plenty of great resources to help
+researchers understand best practices for organizing computational research
+projects, and for developing research software libraries.
+For example, my go-to is [Good Enough Practices in Scientific Computing]
+<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5480810/#sec009title>
+and I am definitely learning a lot from
+<https://merely-useful.github.io/py-rse/py-rse-package-py.html>.
```

Comment:
These links will not work as formatted. i'm actually thinking that we should probably use html links here to open in a new tab if the link is outside of pyopensci.org. the html will look like this dave:

<a href="https://merely-useful.github.io/py-rse/py-rse-package-py.html" target = "_blank">text here describing link</a>

Will you please fix all of the links here that go to an outside page?

Reply:
This one I changed to parentheses instead of angle brackets and removed the line break between the `[title]` and the `(link)` so it should work now.